### PR TITLE
Add basic testing for the tutorial

### PR DESCRIPTION
**Pull request by Tabrizian**
https://github.com/triton-inference-server/server/pull/7176

### CODE REVIEW

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import unittest
+
+from tritonserver import InferenceRequest, Model, Server
+
+
+class IterativeSchedulerTest(unittest.TestCase):
+    def setUp(self):
+        self._server = Server(
+            model_repository="../model_repository",
+            log_info=True,
```
Comment (rmccorm4) [inline Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:37]: Can you add `log_error=True` so we have something in the log if something goes wrong and test fails?

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import time
+import unittest
+
+from tritonserver import InferenceRequest, Model, Server
```
Comment (rmccorm4) [inline Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:30]: Nice use of in-process Python API for testing! This is super simple and clean :rocket:

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py (hunk 1 above)
Comment (Tabrizian) [reply Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:37]: Done

Comment (rmccorm4) [reply Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:37]: These log options are some other things to consider changing the default values on similar to your strict_model_config changes, but leaving that for separate discussions

**Review by rmccorm4 — DISMISSED** [review]

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    def setUp(self):
+        self._server = Server(
+            model_repository="../model_repository",
+            log_info=True,
+            log_error=True,
+            strict_model_config=False,
+        )
+        self._server.start(wait_until_ready=True)
+
+    def _infer_and_verify_request(self, max_tokens, prompt, ignore_eos, model_name):
```
Comment (nnshah1) [inline Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:43]:
```suggestion
    def _infer_and_verify_response(self, max_tokens, prompt, ignore_eos, model_name):
```

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        responses = []
+        response_iterator = model.infer(request)
+        for response in response_iterator:
+            responses.append(response)
+
+        self.assertEqual(max_tokens, len(responses))
+
+    def test_max_tokens(self):
+        self._infer_and_verify_request(
+            32,
```
Comment (nnshah1) [inline Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:59]: suggestion: make 32 a local constant, 

suggestion: make the model name as a parameterized test:

https://realpython.com/pytest-python-testing/#parametrization-combining-tests

https://realpython.com/pytest-python-testing/

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py (hunk 3 above)
Comment (Tabrizian) [reply Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:43]: done

File: Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py (hunk 4 above)
Comment (Tabrizian) [reply Conceptual_Guide/Part_7-iterative_scheduling/tests/test_iterative_scheduler.py:59]: parameterized doesn't work with unittest classes.

**Pull request closed without merging**